### ISSUE TITLE
Cascaded training for lesion segmentation

### ISSUE THREAD

**Issue by naga-karthik** [ivadomed/model_seg_sci#60]
Given that we already have a great SC segmentation model from the region-based strategy, @jcohenadad suggested that a model solely trained for lesion segmentation on images that are cropped around the SC using the predictions from SC seg model could improve performance. This is similar to the approach mentioned in [this paper](https://publications.polymtl.ca/9276/1/2021_Lemay_Automatic_multiclass_intramedullary_spinal_cord.pdf) (see figure 2) which uses ivadomed to do so. 

Compared to the current model that we have (i.e. single model which segments both SC and lesion), this approach would have slightly more complexity because: 

1. Model "A" trained for SC Seg (without sci-paris this time)
2. Get predictions from model "A" and crop the images with SCT
3. Train model "B" for lesion seg on cropped images

We have to repeat these steps at inference time for ideal performance. But, we can still the lesion seg model on uncropped images to see how it performs. 

All things considered, this should be doable, right @valosekj ? 

But one thing that would be difficult is doing steps 1-3 on-the-fly. Even in my conversations with Andreanne (author of the above paper) she said two models had to be used (disjointly) in the cascade architecture. But, if it results in good performance, having another model is justified.

**Comment by naga-karthik** [comment]
We tried the cascased training with 2 models: one for SC segmentation and the second for lesion seg on cropped SC images. The comparison results are in slides 11-13 [here](https://docs.google.com/presentation/d/1fYiy0jXs3wbWg7NCVRqCfYXYPgtASFE4tkUmawxUAPA/edit#slide=id.g288ec4e06c0_0_0). 

TL;DR --> the cascade does not improve the performance drastically (and it has 2 models to be maintained). Hence, we proceed with a single, region-based model trained on full images for SC and lesion segmentation (together)
